src/client/mod.rs

```diff
@@ -1366,7 +1367,10 @@ fn __fetch_group_offsets(req: protocol::OffsetFetchRequest,
                     }
                 }
             }
+
+            topic_map.insert(tp.topic, partition_offsets);
```

Comment:
this - getting rid of the original clone/allocation - is really cool! looking forward to reviewing the whole change. :+1: